### PR TITLE
implementation of database table and entity classes

### PR DESCRIPTION
**Pull request by sepehr9877**

### CODE REVIEW

**Review by nioushamahjoub — APPROVED** [review]

File: src/main/java/com/soen6011/SOEN6011/model/Applications.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToOne;
+
+enum ApplicationStatus{
+	ACCEPTED,REJECTED
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Applications.java:11]: separate lines

File: src/main/java/com/soen6011/SOEN6011/model/Applications.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+@Entity
+public class Applications {
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Applications.java:14]: Application, singular

File: src/main/java/com/soen6011/SOEN6011/model/Applications.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	private Candidates candidates;
+	@OneToOne
+	@JoinColumn(name="joboffer_id")
+	private Joboffers joboffers;
+	public Applications(ApplicationStatus status, Candidates candidates, Joboffers joboffers) {
+		this.status = status;
+		this.candidates = candidates;
+		this.joboffers = joboffers;
+	}
+	public int getId() {
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Applications.java:31]: What is the id?

File: src/main/java/com/soen6011/SOEN6011/model/Educations.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import java.time.LocalDate;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+@Entity
+public class Educations {
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Educations.java:13]: Education, singular

File: src/main/java/com/soen6011/SOEN6011/model/Educations.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+	@GeneratedValue(strategy =GenerationType.IDENTITY)
+	private int id;
+	
+	private String university;
+		private LocalDate start;
+	private LocalDate finish;
+	private String degree;
+	@JoinColumn(name = "resume_id")
+	@ManyToOne
+	private Resume resume;
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Educations.java:25]: Why is there a resume in education?

File: src/main/java/com/soen6011/SOEN6011/model/Educations.java [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+	public Educations(String university, LocalDate start, LocalDate finish, String degree, Resume resume) {
+		super();
+		this.university = university;
+		this.start = start;
+		this.finish = finish;
+		this.degree = degree;
+		this.resume = resume;
+	}
+
+	public int getId() {
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Educations.java:35]: What is the id?

File: src/main/java/com/soen6011/SOEN6011/model/Educations.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+import jakarta.persistence.ManyToOne;
+@Entity
+public class Educations {
+	
+	@Id
+	@GeneratedValue(strategy =GenerationType.IDENTITY)
+	private int id;
+	
+	private String university;
+		private LocalDate start;
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Educations.java:20]: startDate

File: src/main/java/com/soen6011/SOEN6011/model/Educations.java [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+	private LocalDate finish;
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Educations.java:21]: finishDate

File: src/main/java/com/soen6011/SOEN6011/model/Experiences.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+
+@Entity
+public class Experiences {
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Experiences.java:11]: WorkExperience, singular

File: src/main/java/com/soen6011/SOEN6011/model/Joboffers.java [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToOne;
+enum JobOfferStatus{
+ ACCEPTED,REJECTED,PENDING	
+}
+@Entity
+public class Joboffers {
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Joboffers.java:15]: JobOffer, singular

File: src/main/java/com/soen6011/SOEN6011/model/Joboffers.java [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+import java.time.LocalDate;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToOne;
+enum JobOfferStatus{
+ ACCEPTED,REJECTED,PENDING
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Joboffers.java:12]: separate lines

File: src/main/java/com/soen6011/SOEN6011/model/Resume.java [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+import jakarta.persistence.OneToOne;
+@Entity
+public class Resume {
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private int id;
+
+	@OneToOne
+	@JoinColumn(name = "candidate_id")
+	private Candidates candidate;
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Resume.java:23]: Candidate has a Resume, not the other way

File: src/main/java/com/soen6011/SOEN6011/model/Skills.java [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+
+@Entity
+public class Skills {
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/model/Skills.java:11]: Skill, singular

**Comment by MajorTreble** [comment]
No unit tests
No javadoc

**Pull request merged**